examples/nlp/pretraining_BERT.py

```diff
@@ -0,0 +1,481 @@
+yourself, make sure the [MASK] token is among the special tokens you passed during training!
+
+To get the data ready for MLM, we will simply use the `collator` called the
+`DataCollatorForLanguageModeling` provided by the 🤗 Transformers library on our dataset
+that is already ready for the NSP task. The `collator` expects certain parameters.
+We will use the default ones from the original BERT paper in this notebook. The
+`return_tensors='tf'` ensures that we get `tf.Tensor` objects back.
+"""
+
+from transformers import DataCollatorForLanguageModeling
+
+collater = DataCollatorForLanguageModeling(
+    tokenizer=tokenizer, mlm=True, mlm_probability=MLM_PROB, return_tensors="tf"
+)
+
+"""
+Next we define our training set with which we will train our model. Again, 🤗 Datasets
+provides us with the `to_tf_dataset` method which will help us integrate our dataset with
+the `collator` defined above. The method expects certain parameters:
+
+- **columns**: the columns which will serve as our independant variables
+- **label_cols**: the columns which will serve as our labels or dependant variables
+- **batch_size**: our batch size for training
+- **shuffle**: whether we want to shuffle our training dataset
+- **collate_fn**: our collator function
+"""
+
+train = tokenized_dataset.to_tf_dataset(
+    columns=["input_ids", "token_type_ids", "attention_mask"],
+    label_cols=["labels", "next_sentence_label"],
+    batch_size=TRAIN_BATCH_SIZE,
+    shuffle=True,
+    collate_fn=collater,
+)
+
+"""
+## Defining the model
+"""
+
+"""
+To define our model, first we need to define a config which will help us define certain
+parameters of our model architecture. This includes parameters like number of transformer
+layers, number of attention heads, hidden dimension, etc. For this notebook, we will try
+to define the exact config defined in the original BERT paper.
+
+We can easily achieve this using the `BertConfig` class from the 🤗 Transformers library.
+The `from_pretrained()` method expects the name of a model. Here we define the simplest
+model with which we also trained our model, i.e., `bert-base-cased`.
+"""
+
+from transformers import BertConfig
+
+config = BertConfig.from_pretrained(MODEL_CHECKPOINT)
+
+"""
+For defining our model we use the `TFBertForPreTraining` class from the 🤗 Transformers
+library. This class internally handles everything starting from defining our model, to
+unpacking our inputs and calculating the loss. So we need not do anything ourselves except
+defining the model with the correct `config` we want!
+"""
+
+from transformers import TFBertForPreTraining
+
+model = TFBertForPreTraining(config)
+
+"""
+Now we will define our optimizer and compile the model. The loss calculation is handled
+internally and so we need not worry about that!
+"""
+
+optimizer = keras.optimizers.Adam(learning_rate=LEARNING_RATE)
+
+model.compile(optimizer=optimizer, run_eagerly=True)
```

Comment:
Why run eagerly? Typically this is only for debugging.

Reply:
Removed eager execution!